app/controllers/user_profiles_controller.rb

```diff
@@ -59,6 +60,12 @@ def delete_profile_image
     redirect_to controller: 'user_profiles', action: 'show', username: @user.username
   end
 
+  def max_project
+    projects_array = public_courses.map { |course| Wiki.find_by(id: course.home_wiki_id).project }
```

Comment:
This is an N+1 query; it does a SELECT on the wikis table for every course. You can rewrite this so that all the wikis are loaded at once before the `map`.